File: src/main/java/com/indiepost/dto/PostSummaryDto.java
package com.indiepost.dto;

import com.indiepost.enums.PostEnum;
import com.indiepost.model.ImageSet;

import java.io.Serializable;
import java.util.Date;

/**
 * Created by jake on 17. 1. 21.
 */
public class PostSummaryDto implements Serializable {

    private static final long serialVersionUID = 1L;

    private Long id;

    private boolean featured;

    private boolean picked;

    private String title;

    private String excerpt;

    private String displayName;

    private Date publishedAt;

    private ImageSet titleImage;

    private Long titleImageId;

    private PostEnum.Status status;

    private Long categoryId;

    private int commentsCount;

    private int likesCount;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getDisplayName() {
        return displayName;
    }

    public void setDisplayName(String displayName) {
        this.displayName = displayName;
    }

    public ImageSet getTitleImage() {
        return titleImage;
    }

    public void setTitleImage(ImageSet titleImage) {
        this.titleImage = titleImage;
    }

    public boolean isFeatured() {
        return featured;
    }

    public void setFeatured(boolean featured) {
        this.featured = featured;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getExcerpt() {
        return excerpt;
    }

    public void setExcerpt(String excerpt) {
        this.excerpt = excerpt;
    }

    public Date getPublishedAt() {
        return publishedAt;
    }

    public void setPublishedAt(Date publishedAt) {
        this.publishedAt = publishedAt;
    }

    public PostEnum.Status getStatus() {
        return status;
    }

    public void setStatus(PostEnum.Status status) {
        this.status = status;
    }

    public Long getCategoryId() {
        return categoryId;
    }

    public void setCategoryId(Long categoryId) {
        this.categoryId = categoryId;
    }

    public int getCommentsCount() {
        return commentsCount;
    }

    public void setCommentsCount(int commentsCount) {
        this.commentsCount = commentsCount;
    }

    public int getLikesCount() {
        return likesCount;
    }

    public void setLikesCount(int likesCount) {
        this.likesCount = likesCount;
    }

    public Long getTitleImageId() {
        return titleImageId;
    }

    public void setTitleImageId(Long titleImageId) {
        this.titleImageId = titleImageId;
    }

    public boolean isPicked() {
        return picked;
    }

    public void setPicked(boolean picked) {
        this.picked = picked;
    }
}

File: src/main/kotlin/com/indiepost/dto/post/PostSummaryDto.kt
package com.indiepost.dto.post

import com.indiepost.dto.Highlight
import com.indiepost.dto.ImageSetDto
import com.indiepost.enums.Types.PostStatus
import java.io.Serializable
import java.time.Instant

/**
 * Created by jake on 17. 1. 21.
 */
open class PostSummaryDto : Serializable {

    var id: Long? = null

    var isFeatured: Boolean = false

    var isPicked: Boolean = false

    var isSplash: Boolean = false

    var title: String? = null

    var excerpt: String? = null

    var displayName: String? = null

    var publishedAt: Instant? = null

    var modifiedAt: Instant? = null

    var isShowLastUpdated: Boolean = false

    var titleImage: ImageSetDto? = null

    var titleImageId: Long? = null

    var status: PostStatus? = null

    var primaryTag: String? = null

    var commentsCount: Int = 0

    var lastRead: Instant? = null

    var bookmarked: Instant? = null

    var isInteractionFetched: Boolean = false

    var highlight: Highlight? = null

}
